cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorCustomCertMockTest.java

```diff
@@ -124,16 +127,28 @@ public void setup() {
             .addToData("foo", "bar")
             .build();
         client.secrets().inNamespace(namespace).create(secret);
-        ResourceOperatorSupplier supplier = new ResourceOperatorSupplier(vertx, client, mock(ZookeeperLeaderFinder.class),
-                mock(AdminClientProvider.class), mock(ZookeeperScalerProvider.class),
-                mock(MetricsProvider.class), new PlatformFeaturesAvailability(false, KubernetesVersion.V1_20), FeatureGates.NONE, 10000);
-
-        operator = new MockKafkaAssemblyOperator(vertx, new PlatformFeaturesAvailability(false, kubernetesVersion),
+        PlatformFeaturesAvailability platformFeaturesAvailability = new PlatformFeaturesAvailability(false, kubernetesVersion);
+        ResourceOperatorSupplier supplier = new ResourceOperatorSupplier(
+                vertx,
+                client,
+                mock(ZookeeperLeaderFinder.class),
+                mock(AdminClientProvider.class),
+                mock(ZookeeperScalerProvider.class),
+                mock(MetricsProvider.class),
+                platformFeaturesAvailability,
+                FeatureGates.NONE,
+                10000,
+                KubernetesRestartEventPublisher.createPublisher(client, "op", platformFeaturesAvailability.hasEventsApiV1())
+        );
```

Comment:
Pulled the PFA out into a variable, as the event publisher has a dependency on it.